nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/canvas/canvas.component.ts

```diff
@@ -759,10 +760,45 @@ export class Canvas implements OnInit, OnDestroy {
                 return copyResponse;
             }
 
-            // attempting to paste something other than CopyResponseEntity
+            // check to see if this is a FlowDefinition
+            const maybeFlowDefinition: any = JSON.parse(json);
+            const isFlowDefinition = this.nifiCommon.isDefinedAndNotNull(maybeFlowDefinition.flowContents);
+            if (isFlowDefinition) {
+                // make sure the flow has some components
+                const flowHasCopiedContent = Object.entries(maybeFlowDefinition.flowContents).some((entry) => {
+                    return supportedKeys.includes(entry[0]) && Array.isArray(entry[1]) && entry[1].length > 0;
```

Comment:
```suggestion
                    return supportedKeys.includes(entry[0]) && Array.isArray(entry[1]);
```

Thoughts on relaxing this condition? With it as-is, someone could export an empty flow definition and be rejected when pasting calling it an `incompatible format`.

Reply:
Yeah, i did consider that originally but decided against it just to be consistent with the logic for verifying the CopyResponseEntity. However, these are a bit different use cases so I can make that change.